File: src/main/java/com/tobe/healthy/gym/application/GymService.java
package com.tobe.healthy.gym.application;

import java.util.List;
import java.util.stream.Collectors;

import org.springframework.stereotype.Service;
import org.springframework.transaction.annotation.Transactional;

import com.tobe.healthy.gym.presentation.dto.out.GymResult;
import com.tobe.healthy.gym.presentation.dto.out.TrainersByGymResult;
import com.tobe.healthy.gym.repository.GymRepository;
import com.tobe.healthy.member.repository.MemberRepository;

import lombok.RequiredArgsConstructor;

@Service
@Transactional(readOnly = true)
@RequiredArgsConstructor
public class GymService {

	private final MemberRepository memberRepository;
	private final GymRepository gymRepository;

	public List<GymResult> findAllGym() {
		return gymRepository.findAll().stream()
			.map(GymResult::from)
			.collect(Collectors.toList());
	}

	public List<TrainersByGymResult> findAllTrainersByGym(Long gymId) {
		return memberRepository.findAllTrainerByGym(gymId).stream()
			.map(TrainersByGymResult::from)
			.collect(Collectors.toList());
	}
}

File: src/main/kotlin/com/tobe/healthy/gym/application/GymService.kt
package com.tobe.healthy.gym.application

import com.tobe.healthy.gym.domain.dto.out.GymResult
import com.tobe.healthy.gym.domain.dto.out.TrainersByGymResult
import com.tobe.healthy.gym.repository.GymRepository
import com.tobe.healthy.member.repository.MemberRepository
import org.springframework.stereotype.Service
import org.springframework.transaction.annotation.Transactional

@Service
@Transactional(readOnly = true)
class GymService(
    private val memberRepository: MemberRepository,
    private val gymRepository: GymRepository,
) {
    fun findAllGym(): List<GymResult> {
        return gymRepository.findAll().map { GymResult.from(it) }
    }

    fun findAllTrainersByGym(gymId: Long): List<TrainersByGymResult?> {
        return memberRepository.findAllTrainerByGym(gymId).map { TrainersByGymResult.from(it) }
    }
}